test/reporters/xunit.spec.js

```diff
@@ -194,15 +194,13 @@ describe('XUnit reporter', function() {
       it('should call write with line', function() {
         stdoutWrite = process;
         process = false; // eslint-disable-line no-native-reassign, no-global-assign
-        var cachedConsoleLog = console.log;
-        console.log = function(string) {
-          stdout.push(string);
-        };
 
+        var savedConsoleLog = reporters.Base.consoleLog;
+        reporters.Base.consoleLog = stdout.push.bind(stdout);
         var xunit = new XUnit({on: function() {}, once: function() {}});
         xunit.write.call({fileStream: false}, expectedLine);
+        reporters.Base.consoleLog = savedConsoleLog;
 
-        console.log = cachedConsoleLog;
         process = stdoutWrite; // eslint-disable-line no-native-reassign, no-global-assign
         expect(stdout[0], 'to be', expectedLine);
       });
```

Comment:
This is what I'm using...
Achieves equivalent test without messing with `process` object and produces no "noise" output...

```js
    describe('when output directed to console', function() {
      var xunit;

      it("should call 'console.log' with line", function() {
        xunit = new XUnit(runner);

        var fakeThis = {preferConsole: true};
        Object.setPrototypeOf(fakeThis, XUnit.prototype);
        var consoleLog = sinon.stub(console, 'log');
        xunit.writeln.call(fakeThis, expectedLine);
        consoleLog.restore();

        expect(consoleLog.calledWith(expectedLine), 'to be true');
      });
    });
```

Reply:
Of course the user could have used this same technique rendering this issue unnecessary.

Reply:
But the logic inside `xunit` reporter requires `process` to have been tampered with doesnt it? 
The assertion is on the logging but the test setup will have to mess with `process` wont it?

```javascript
  if (this.fileStream) {
    // ... not this
  } else if (typeof process === 'object' && process.stdout) {
    // ... not this
  } else {
   // ... want to call this
  }
```

Happy to be corrected here if ive missed something.

Reply:
You're right, of course.

I'm too far ahead of myself here; I adjusted that part of the code to make XUnit more easily configurable (and testable)...

Reply:
But the scheme for stubbing `console.log` (assuming XUnit instance was thus configured) is demonstrated.